src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNILoadHandle.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading;
+
+namespace System.Data.SqlClient.SNI
+{
+    /// <summary>
+    /// Global SNI settings and status
+    /// </summary>
+    internal class SNILoadHandle
+    {
+        static public SNILoadHandle SingletonInstance = new SNILoadHandle();
+
+        private ThreadLocal<SNIError> _lastError = new ThreadLocal<SNIError>(() => { return new SNIError(SNIProviders.INVALID_PROV, 0, 0, "No SNI Error has been reported yet."); });
```

Comment:
I'm not sure what's exposed publically and what's not... does this string need to be in the .resx?